crates/miden-objects/src/note/note_tag.rs

```diff
@@ -114,7 +120,7 @@ impl NoteTag {
                 let high_bits = high_bits & 0xffff0000;
 
                 // Set the local execution tag in the two most significant bits.
-                Ok(Self(high_bits | LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED))
+                Ok(Self::LocalAccount(high_bits | LOCAL_EXECUTION_WITH_ALL_NOTE_TYPES_ALLOWED))
```

Comment:
Since we us enum variants to represent the high bits, I'm not sure merging high bits is necessary here. This applies to all other variants too.